w3- code-challenges.test.js

```diff
@@ -0,0 +1,221 @@
+// Add appropriate dependencies: $ yarn add jest
+
+// Reminder: The test will call your function
+// Run the file with the following command: $ yarn jest
+
+
+//------------------------------------ Assessment Start ------------------------------------
+
+// jest test template:
+// describe ("nameOfFunction", () => {
+//     it("explains what it takes and return expectation in plain English as this is for the programmer", () => {
+//         expect(nameOfFunction()).toEqual(expected return)
+//     })
+// })
+
+
+// object template:
+// const objectName = { key: value }
+
+// --------------------1) Create a function that takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence.
+
+// a) Create a test with expect statements for each of the variables provided.
+
+const fibLength1 = 6
+// Expected output: [1, 1, 2, 3, 5, 8]
+
+const fibLength2 = 10
+// Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
+
+
+//------------------------------------ My Test Code ------------------------------------
+
+// pseudo code:
+// Using jest test template to test a function "orderedFibs"
+// it "takes a number > 2 and returns array of that length with the numbers of the fibonacci sequence as values"
+// expect(orderedFibs(fibLength1)).toEqual([1, 1, 2, 3, 5, 8])
+// expect(orderedFibs(fibLength2)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+// input: since there isn't a function to test, there is no input and I expect a fail (red)
+// output: should output a ReferenceError: orderedFibs is not defined
+
+describe ("orderedFibs", () => {
+  it("takes a number > 2 and returns array of that length with the numbers of the fibonacci sequence as values", () => {
+    expect(orderedFibs(fibLength1)).toEqual([1, 1, 2, 3, 5, 8])
+    expect(orderedFibs(fibLength2)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+    })
+  })
+
+// output ReferenceError:  FAIL  ./w3- code-challenges.test.js
+// orderedFibs is not defined
+
+// b) Create the function that makes the test pass.
+
+// pseudo code:
+// what's needed? A function that takes a number > 2 parameter and returns an array with a length of the argument passed
+// the fibonnaci sequence is a sequence whereby each new number is the sum of the previous two numbers 0, 1, 1, 2, 3, 5, 8...
+// set up three variable to rotate and update the current fibonacci number
+// input: number > 2
+// output: array of length === input number
+// expected result:  PASS
+ 
+const orderedFibs = (fibArrLength) => {
+  const fibArray = []
+  let firstFib = 0
+  let secondFib = 1
+  let nextFib = 1
+
+  if (fibArrLength > 2) {
+    for (let i = 0; i < fibArrLength; i ++){
+      fibArray[i] = nextFib
+      nextFib = firstFib + secondFib
+      firstFib = secondFib
+      secondFib = nextFib 
+    }
+  }
+  return fibArray
+}
+
+// output PASS  ./w3- code-challenges.test.js
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+
```

Comment:
Excellent Borja!  Your pseudocode and test are great.  I like the way you set up your function. 
Just for fun, see if you can refactor with higher order functions. 😬